CMakeLists.txt

```diff
@@ -3,9 +3,29 @@ project(polygames)
 
 set(CMAKE_CXX_STANDARD 17)
 
-set(CMAKE_CXX_FLAGS_DEBUG "-g -O0 -Wall -Wextra")
-set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-g -O3 -ffast-math")
-set(CMAKE_CXX_FLAGS_RELEASE "-O3 -ffast-math")
+OPTION(USE_MOLD "Use mold linker if available" ON)
+
+if(USE_MOLD)
+  find_program(MOLD_FOUND mold)
+  if(MOLD_FOUND)
+    message(STATUS "Using mold linker: mold")
+    set(LINK_FLAGS "-fuse-ld=mold")
+  else()
+    message(STATUS "mold linker not found, using default linker `ld`")
+    set(LINK_FLAGS "-fuse-ld=lld")
+  endif()
+endif()
+
+
+set(FLAGS_WARNINGS "-Wall -Wextra -Werror -Wno-unused-variable -Wno-unused-parameter -Wno-unused-value -Wno-deprecated-declarations")
```

Comment:
TODO remove `-Wno-unused-parameter` and `-Wno-unused-value`.


TODO use Clang which has more interesting warnings.